### PR TITLE
feat: kadena inclusion proof

### PR DESCRIPTION
**Pull request by tchataigner**
This PR adds the logic for verifying an SPV proof fetched from a `/spv` endpoint of a Chainweb node.

## Changelog

- Implement the types for `Spv`and `MerkleProof`, along with some serde methods to decode them from the API and send them over to the proof server
- Add an `SpvProver` structure to handle proof generation and verification
- Revamp the proof server to handle the new program
- Add a test asset containing a valid SPV for testing purposes

### CODE REVIEW

**Comment by tchataigner** [comment]
The cycle count regression CI will be broken as I have renamed the test to fit the LC, change has been made in the workflow file in this PR so that it leverages the correct ones.

File: kadena/core/src/crypto/hash/sha512.rs [hunk 1]
```diff
@@ -42,8 +42,26 @@ pub const fn tag_bytes(tag: u16) -> [u8; TAG_BYTES_LENGTH] {
 ///
 /// The hash of the given data.
 pub fn hash_tagged_data(tag: u16, bytes: &[u8]) -> Result<HashValue, CryptoError> {
+    let input = [&tag_bytes(tag), bytes].concat();
+    hash_leaf(&input)
+}
+
+/// Hash the given data as a Merkle leaf using the SHA-512 hash function
+/// following the model of the Chainweb Merkle tree. To do so, it prepends the byte
+/// `0x0` to the data and a given tag.
```
Comment (wwared) [inline kadena/core/src/crypto/hash/sha512.rs]: Is this docstring accurate? It doesn't look like this function adds any tags

File: kadena/core/src/merkle/spv.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        bytes.extend_from_slice(&self.chain().to_be_bytes());
+
+        // Add algorithm length
+        let algorithm_bytes = self.algorithm().as_bytes();
+        bytes.extend_from_slice(&(algorithm_bytes.len() as u32).to_be_bytes());
+
+        // Add algorithm
+        bytes.extend_from_slice(algorithm_bytes);
+
+        // Add subject length
```
Comment (wwared) [inline kadena/core/src/merkle/spv.rs]: nit:
```suggestion
        // Add subject
```

**Comment by tchataigner** [comment]
It has to be noted that @wwared also raised some concerns about the `kadena-lc-core::merkle` module using big endian to serialized numbers while other modules use little endian. I have used big endian in the `kadena-lc-core::merkle` because the `MerkleProof` serde methods (based on code on Chainweb side) also leverage big endian.

Thus, we have a "standard" of development with big endian being used on merkle related data while the rest is encoded with little endian.

**Review by wwared — APPROVED** [review]
The different endianess used in some of the encodings is a bit weird, but as far as I can tell it's all consistent (i.e. both encode and decode always use the same endianess) and properly documented, so not a big deal.

**Pull request merged**